channeld/channeld.c

```diff
@@ -202,6 +207,52 @@ static void billboard_update(const struct peer *peer)
 		       announce_status, shutdown_status);
 }
 
+/* used in billboard_update to say how many blocks to go! */
+static char *get_funding_depth(const struct peer *peer)
+{
+	status_trace("Please wait. We are trying to get depth...");
+	u8 *msg, *reply;
+	u32 minimum_depth, funding_depth;
+	char *infor = tal_arr(tmpctx, char, 0);
+	struct bitcoin_txid txid;
+
+	msg = towire_channel_funding_depth(tmpctx, &peer->channel->funding_txid);
+
+	/* wait for the reply, this loop may cost some minutes */
+	reply = master_wait_sync_reply(tmpctx,
+				       cast_const(struct peer *, peer),
```

Comment:
I am not an expert, but I cannot find any other such use of `cast_const` in the clightning.
`peer` is already of type `const struct peer *`, so we won't use it to modify `peer`.

Reply:
[static void billboard_update(const struct peer *peer)](https://github.com/ElementsProject/lightning/blob/3e4a3bafc3db443c00a1bb16f95c0f0ca69850d4/channeld/channeld.c#L161) used the `const struct peer *`,  
but  [static u8 *master_wait_sync_reply(const tal_t *ctx, struct peer *peer, const u8 *msg, enum channel_wire_type replytype)](https://github.com/ElementsProject/lightning/blob/3e4a3bafc3db443c00a1bb16f95c0f0ca69850d4/channeld/channeld.c#L761) used the `struce peer *`,  so I need change `const struct peer *` to `struct peer *` in`get_funding_depth()`.